### PR TITLE
AS-581: allow null values when property not required

### PR DESCRIPTION
**Pull request by wheelsandcogs**
Don't throw validation errors if `null` values are passed for properties which are not required by the schema.

### CODE REVIEW

File: schema/authorisation.js [hunk 1]
```diff
@@ -9,18 +9,18 @@ class Authorisation extends BaseModel {
   static get jsonSchema() {
     return {
       type: 'object',
-      required: ['type', 'method', 'description'],
+      required: ['type', 'method', 'description', 'establishmentId'],
```
Comment (wheelsandcogs) [inline schema/authorisation.js]: I missed `establishmentId` as a required field from authorisation (having a memory failure if this was intentional or not?)

Comment (lennym) [reply schema/authorisation.js]: It's definitely required. Good catch.

Comment (joefitter) [reply schema/authorisation.js]: Yeah establishmentId should be required

File: schema/establishment.js [hunk 2]
```diff
@@ -12,31 +12,31 @@ class Establishment extends BaseModel {
...
-        rehomes: { type: 'boolean' },
-        conditions: { type: 'boolean' },
-        'created_at': { type: 'string' },
-        'updated_at': { type: 'string' }
+        procedure: { type: ['boolean', null] },
+        breeding: { type: ['boolean', null] },
+        supplying: { type: ['boolean', null] },
+        killing: { type: ['boolean', null] },
+        rehomes: { type: ['boolean', null] },
+        conditions: { type: ['boolean', null] },
```
Comment (lennym) [inline schema/establishment.js]: I don't think the boolean types should be nullable. They can be missing completely, but shouldn't ever have null value.

**Comment by lennym** [comment]
On further thought I think we should explicitly reject `created_at` and `updated_at` as they should _always_ be set automatically, and so any attempt to set them explicitly should be rejected.

**Comment by wheelsandcogs** [comment]
What about importing existing records? Would you want the original creation date if we have it?

**Comment by lennym** [comment]
🤔 You raise a very valid point. Leave it for now.

File: schema/place.js [hunk 3]
```diff
@@ -13,10 +13,10 @@ class Place extends BaseModel {
       required: ['site', 'name', 'suitability', 'holding', 'establishmentId'],
       additionalProperties: false,
       properties: {
-        id: { type: 'string' },
-        'migrated_id': { type: 'string' },
+        id: { type: ['string', 'null'] },
```
Comment (lennym) [inline schema/place.js]: `id` definitely cannot be null.

**Comment by lennym** [comment]
Looks like you've generally made `id` a nullable property, which it definitely isn't. It can be _undefined_ (i.e. not present as a property at all) but if it's defined then it needs to have a string value.

File: schema/profile.js [hunk 4]
```diff
@@ -14,19 +14,19 @@ class Profile extends BaseModel {
       properties: {
         id: { type: 'string' },
         'migrated_id': { type: 'string' },
```
Comment (joefitter) [inline schema/profile.js]: migrated_id will be nullable, as not all models will have been migrated

Comment (joefitter) [reply schema/profile.js]: ignore me, I'm obviously drunk

**Pull request merged**